backend/coreapp/github.py

```diff
@@ -133,3 +135,93 @@ def login(request: Request, oauth_code: str) -> "GitHubUser":
         request.session["profile_id"] = profile.id
 
         return gh_user
+
+class GitHubRepo(models.Model):
+    owner = models.CharField(max_length=100)
+    repo = models.CharField(max_length=100)
+    branch = models.CharField(max_length=100, default="master", blank=False)
+    gh_user = models.ForeignKey(GitHubUser, null=False, help_text="Must have admin permission on the repo", on_delete=models.PROTECT)
+    is_pulling = models.BooleanField(default=False)
+    last_pulled = models.DateTimeField(blank=True, null=True)
+
+    class Meta:
+        verbose_name = "GitHub repo"
+        verbose_name_plural = "GitHub repos"
+
+    # TODO: make this async
+    def pull(self):
+        if self.is_pulling:
+            raise Exception("Already pulling")
+
+        self.is_pulling = True
+        self.save()
+
+        try:
+            repo_dir = self.get_dir()
+            remote_url = f"https://github.com/{self.owner}/{self.repo}"
+
+            if repo_dir.exists():
+                subprocess.run(["git", "remote", "set-url", "origin", remote_url], cwd=repo_dir)
```

Comment:
might want to look at something like https://www.pygit2.org/ if this moves out of MVP

Reply:
Personally I'm not entirely certain why this would be better than just using the shell git; we don't need to do very much with git

Reply:
because if we do it in python, we don't have to deal with the messiness of external processes and interpreting error codes / stderr / stdout. we just have a simple python object to work with. in general I think avoiding raw process calls from python is cleaner and better design

Reply:
<img width="758" alt="image" src="https://user-images.githubusercontent.com/9429556/154088333-ba3ec981-5caa-4059-98e1-f8027a679529.png">